### PR TITLE
Prevent unverified email changes

### PR DESCRIPTION
**Pull request by jamiecobbett**
Normal users shouldn't be able to directly change their own email address - they have to confirm the change via a link sent to the new email address (a notification is also sent to their old address).

Admins have been able to make email changes without requiring the user to confirm for some time. 

We accidentally gave normal users access to/a link to the admin-style page for themselves, which allows them to change their own email without confirmation.

By preventing their access to the full fledged edit page, we also mitigate some usability problems for normal users - it included various fields and links which would lead to the user seeing errors.

cc @jennyd @bishboria

### CODE REVIEW

**Comment by bishboria** [comment]
:+1:

**Comment by jamiecobbett** [comment]
Rebased on top of changes on master.

File: spec/policies/user_policy_spec.rb [hunk 1]
```diff
@@ -64,7 +64,8 @@
     end
   end
 
-  (user_management_actions - [:event_logs?]).each do |permission_name|
+  self_management_actions = [:edit_email_or_passphrase?, :update_passphrase?, :cancel_email_change?, :resend_email_change?]
```
Comment (jennyd) [inline spec/policies/user_policy_spec.rb]: Shouldn't `:update_email?` also be in this list?

Comment (jamiecobbett) [inline spec/policies/user_policy_spec.rb]: Yes, rebased.

File: test/controllers/users_controller_test.rb [hunk 2]
```diff
@@ -56,7 +56,7 @@ def change_user_password(user_factory, new_password)
       end
 
       should "show the unconfirmed_email" do
-        get :edit, id: @user.id
+        get :edit_email_or_passphrase, id: @user.id
```
Comment (jennyd) [inline test/controllers/users_controller_test.rb]: The context description needs updating for this change.

Comment (jamiecobbett) [inline test/controllers/users_controller_test.rb]: Well spotted - rebased.

**Pull request merged**